src/idiom173ex.fz

```diff
@@ -0,0 +1,30 @@
+# Example that returns the formatted string instead of printing it directly
+
+format(x T : integer)  String =>
+
+    str:= x.as_string
+
+    len := str.byte_length
+
+    first :=
+        if x < T.zero && ((len%3) = 1) then 4   # avoid seperator right after minus
+        else if (len%3) > 0 then len%3 else 3   # determin size of first group
```

Comment:
The last `else` should go to a new line and some parentheses are not needed. Also, it is IMHO good to align values that are related vertically, which makes the code easier to read for developers with visual perception.
```suggestion
        if x < T.zero && len%3 = 1 then 4      # avoid separator right after minus
        else if len%3 > 0          then len%3  # determine size of first group
        else                            3
```